sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala

```diff
@@ -52,6 +52,11 @@ import org.apache.spark.unsafe.types.UTF8String
       > SELECT _FUNC_('12,454.8-', '99G999D9S');
        -12454.8
   """,
+  note = """
+    ToNumber has better compatibility with number format based on the underlying implementation.
```

Comment:
```suggestion
    `to_number` has better compatibility with number format based on the underlying implementation.
```

Let's use `to_number` too since this is user-facing docs.